shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -844,7 +911,7 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
             res
           }
       case v @ ValidVar(name) =>
-        val ty = ctx.get(name).fold(err("identifier not found: " + name, term.toLoc): ST) {
+        val ty = ctx.get(name).fold(err((if (ctx.inQuote) "unbound quoted variable: " else "identifier not found: ") + name, term.toLoc): ST) {
```

Comment:
Why would you change the error like this when it's in a quote? I.e., one says "unbound" and "identifier", and the other "not found" and "variable". Do you just want to introduce inconsistencies for no reason?

Reply:
We previously discussed it. We need another error message for unbound identifiers in quotes instead of just "identifier not found".

Reply:
What? We never discussed using inconsistent terminology.

The remark I was making when we discussed these errors is that it would be more helpful if we did _not_ have an error like this (whatever the phrasing) when the variable is in lexical scope but at the wrong level. It seems you completely missed my point.

Reply:
I suggest just removing this change and moving on, for now.